src/watertap_contrib/seto/costing/units/chemical_softening_zo.py

```diff
@@ -0,0 +1,968 @@
+                )
+                ** chem_soft.recarb_basin_op_exp_2
+            )
+            + (
+                chem_soft.recarb_basin_op_coeff_3
+                * (
+                    pyunits.convert(
+                        blk.unit_model.CO2_first_basin
+                        + blk.unit_model.CO2_second_basin,
+                        to_units=pyunits.lb / pyunits.day,
+                    )
+                )
+                + chem_soft.recarb_basin_op_constant
+            )
+        )
+    )
+
+    # Lime feed
+    blk.lime_feed_op_cost_constraint = Constraint(
+        expr=blk.lime_feed_op_cost
+        == (
+            chem_soft.lime_feed_system_op_coeff
+            * (
+                pyunits.convert(
+                    blk.unit_model.CaO_dosing, to_units=pyunits.lb / pyunits.day
+                )
+            )
+            ** chem_soft.lime_feed_system_op_exp
+        )
+    )
+
+    # Lime sludge management
+    blk.lime_sludge_mngt_op_cost_constraint = Constraint(
+        expr=blk.lime_sludge_mngt_op_cost
+        == (
+            pyunits.convert(
+                blk.unit_model.sludge_prod, to_units=pyunits.tons / pyunits.year
+            )
+            * chem_soft.sludge_disposal_cost
+        )
+    )
+
+    # Admin cost
+    blk.admin_op_cost_constraint = Constraint(
+        expr=blk.admin_op_cost
+        == (
+            chem_soft.admin_op_coeff
+            * (
+                (
+                    pyunits.convert(
+                        blk.unit_model.properties_in[0].flow_vol,
+                        to_units=pyunits.megagallon / pyunits.day,
+                    )
+                )
+                ** chem_soft.admin_op_exp
+            )
+        )
+    )
+
+    # Sum of all capital costs
+    blk.capital_cost_constraint = Constraint(
+        expr=blk.capital_cost
+        == blk.mix_tank_capital_cost
+        + blk.floc_tank_capital_cost
+        + blk.sed_basin_capital_cost
+        + blk.recarb_basin_capital_cost
+        + blk.recarb_basin_source_capital_cost
+        + blk.lime_feed_system_capital_cost
+        + blk.admin_capital_cost
+    )
+
+    # Sum of all operational costs
+
+    if blk.unit_model.config.silica_removal:
+        blk.fixed_operating_cost_constraint = Constraint(
+            expr=blk.fixed_operating_cost
+            == blk.mix_tank_op_cost
+            + blk.floc_tank_op_cost
+            + blk.sed_basin_op_cost
+            + blk.recarb_basin_op_cost
+            + blk.lime_feed_op_cost
+            + blk.lime_sludge_mngt_op_cost
+            + blk.admin_op_cost
+        )
+    else:
+        blk.fixed_operating_cost_constraint = Constraint(
+            expr=blk.fixed_operating_cost
+            == blk.mix_tank_op_cost
+            + blk.floc_tank_op_cost
+            + blk.sed_basin_op_cost
+            + blk.recarb_basin_op_cost
+            + blk.lime_feed_op_cost
+            + blk.lime_sludge_mngt_op_cost
+            + blk.admin_op_cost
+        )
```

Comment:
I don't see the difference between these constraints. Is the conditional needed?

Reply:
I removed the conditional. Originally, MgCl2 was included as an operational cost which was in the else function. But MgCl2 is not a cost flow.